src/yupSchema.js

```diff
@@ -73,13 +106,57 @@ const yupSchemas = {
       .oneOf(options, ({ value }) => {
         return `The option ${JSON.stringify(value)} is not valid.`;
       }),
-  date: string()
-    .nullable()
-    .trim()
-    .matches(
-      /(?:\d){4}-(?:0[1-9]|1[0-2])-(?:0[1-9]|1[0-9]|2[0-9]|3[0-1])/,
-      `Must be a valid date in ${DEFAULT_DATE_FORMAT.toLocaleLowerCase()} format. e.g. ${todayDateHint}`
-    ),
+  date: ({ minDate, maxDate }) => {
+    if (minDate && maxDate) {
+      return string()
+        .nullable()
+        .trim()
+        .matches(
+          /(?:\d){4}-(?:0[1-9]|1[0-2])-(?:0[1-9]|1[0-9]|2[0-9]|3[0-1])/,
+          `Must be a valid date in ${DEFAULT_DATE_FORMAT.toLocaleLowerCase()} format. e.g. ${todayDateHint}`
+        )
+        .test('minDate', `The date must be ${minDate} or after`, (value) =>
+          validateMinDate(value, minDate)
+        )
+        .test('maxDate', `The date must be ${maxDate} or before`, (value) =>
```

Comment:
nit: Can you add a period (.) at the end of each sentence? I know we need to add it too in other error messages, but let's leave that for another PR.

Reply:
Done